tests/features/joinup_seo/news.feature

```diff
@@ -0,0 +1,53 @@
+      | title           | headline                    | logo     | body                    | created                         | publication date                | changed                         | state     | author            | collection                 |
+      | Joinup SEO news | Headline of Joinup SEO news | logo.png | Body of Joinup SEO news | Sun, 01 Dec 2019 13:00:00 +0100 | Wed, 25 Dec 2019 13:00:00 +0100 | Wed, 01 Jan 2020 13:00:00 +0100 | validated | Joinup SEO author | Joinup SEO news collection |
+
+    When I visit the "Joinup SEO news" news
+    Then the metatag json should be attached in the page
+    And 1 metatag graph of type "NewsArticle" should exist in the page
+    And the metatag graph of the item with "name" "Joinup SEO news" should have the following properties:
+      | property         | value                                                                 |
+      | @type            | NewsArticle                                                           |
+      | headline         | Headline of Joinup SEO news                                           |
+      | description      | Body of Joinup SEO news                                               |
+      | datePublished    | Wed, 25/12/2019 - 13:00                                               |
+      | dateModified     | Wed, 01/01/2020 - 13:00                                               |
+      # $base_url$ will be replaced with the base url of the website.
+      | mainEntityOfPage | $base_url$/collection/joinup-seo-news-collection/news/joinup-seo-news |
+    And the metatag graph of the item with "name" "Joinup SEO news" should have the following "image" properties:
+      | property             | value       |
+      | @type                | ImageObject |
+      | representativeOfPage | True        |
+      # The test files are assigned a random name so it is very hard to assert the real url. However, when the file is
+      # found and added to the page, the dimensions are added as well. Asserting the dimensions also asserts the url in
+      # a way.
+      | width                | 377         |
+      | height               | 139         |
+    And the metatag graph of the item with "name" "Joinup SEO news" should have the following "author" properties:
+      | property | value      |
+      | @type    | Person     |
+      | name     | Kurk Smith |
+    And the metatag graph of the item with "name" "Joinup SEO news" should have the following "publisher" properties:
+      | property | value        |
+      | @type    | Organization |
+      | name     | Joinup       |
+      | url      | $base_url$/  |
+
+    When I click "Keep up to date"
+    Then I should see the "Joinup SEO news" tile
+    # No metatags are defined for the keep up to date page.
+    # No metatags JSON in general means also that the entity metatags of the news item
+    # is also not attached when the tile is present.
+    And the metatag json should not be attached in the page
```

Comment:
Yeah I was wondering about this, i know that in earlier metadata implementations (RDFa possibly) Drupal was outputting the data also in teasers etc. Is this the current best practice, or defined in the functional specification?

Reply:
I don't know if there is a "best practice" to generally place any attach JSON to teasers, I guess it depends. What I _do_ know (is mentioned on the Google instructions) is that google suggests that listing pages and search results are not listing tags.
For example, you don't want to search `test` in Google and find a result from the `Solution A` page because it seems that distribution `test` is part of it but when visiting the page, there is no `test` distribution because the page uses a pager and `test` is now on the 15th page.
Same goes for listings like the "Collections" listing page which the `Collection A` is part of the results but the page is not really about the `Collection A` but a listing of collections in general.
That is why I thought that these should not be part of a page where entities are listed as tiles.

Reply:
OK that makes a lot of sense, I agree that we should not output them in the listings. Thanks for enlightening me!